src/node_file.cc

```diff
@@ -1929,6 +1929,58 @@ static void MKDir(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void MKDirSync(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  const int argc = args.Length();
+  CHECK_GE(argc, 3);
+
+  BufferValue path(env->isolate(), args[0]);
+  CHECK_NOT_NULL(*path);
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemWrite, path.ToStringView());
+
+  CHECK(args[1]->IsInt32());
+  const int mode = args[1].As<Int32>()->Value();
+
+  CHECK(args[2]->IsBoolean());
+  bool mkdirp = args[2]->IsTrue();
+  
+  int err;
+  FS_SYNC_TRACE_BEGIN(rmdir);
+
+  if (mkdirp) {
+    FSReqWrapSync req_wrap_sync;
+
+    err = MKDirpSync(env->event_loop(), &(req_wrap_sync.req), *path, mode,
+                       nullptr);
+    if (err == 0 &&
+        !req_wrap_sync.continuation_data()->first_path().empty()) {
+      Local<Value> error;
+      std::string first_path(req_wrap_sync.continuation_data()->first_path());
+      FromNamespacedPath(&first_path);
+      MaybeLocal<Value> path = StringBytes::Encode(env->isolate(),
+                                                    first_path.c_str(),
+                                                    UTF8, &error);
+      if (path.IsEmpty()) {
+        // TODO: throw a good exception here. somehow use error? (without ctx arg)
+        return env->ThrowUVException(err, "rmdir");
```

Comment:
If error exists, you can throw error local object. One of my newer pull requests uses this.